numpy/_core/tests/test_umath.py

```diff
@@ -4818,18 +4818,22 @@ class BadArr1(np.ndarray):
         def __array_finalize__(self, obj):
             # The outer call reshapes to 3 dims, try to do a bad reshape.
             if self.ndim == 3:
-                self.shape = self.shape + (1,)
```

Comment:
I think for these cases where one mimics what real code would/should do, I would use `self._set_shape(...)` rather than catch the warnings - here one really has no choice.